### DISCUSSION TITLE
address /-.example.com/ 中“-”代表什么意思呢？

### DISCUSSION THREAD

**Discussion by Mosney** [pymumu/smartdns#1602] [Q&A] [answered]
https://pymumu.github.io/smartdns/configuration/
```
address /[*|-.]domain/[ip1[,ip2,...]|-|-4|-6|#|#4|#6]
- 表示忽略
# 表示返回 SOA
4 表示 IPv4
6 表示 IPv6
* 开头表示通配
- 开头表示主域名
* 和 - 只能在域名开头，其他位置不生效。
```

不明白这个 `- 开头表示主域名` 是什么意思，如果是“减”去一切子域的意思的话，
` address /-.example.com/`
与
` address /example.com/`
有什么区别？

**Comment by Mosney** [comment]
以及文档中给出的最后三个示例非常具有迷惑性
```
address /*-a.example.com/
address /*.example.com/
address /-.example.com/
```
第一个容易被误以为*-可以联用，
这三个示例按我的理解后面应该跟一个IP或者`-|-4|-6|#|#4|#6`任意一种，但是示例后面是空的，搞不太明白这三个示例配置是什么意思

**Comment by PikuZheng** [comment, marked as answer]
（重要）支持使用*区分主域名和子域名
匹配 example.com 和 *.example.com
example.com

匹配 *.example.com，不匹配 example.com
*.example.com

匹配 example.com，不匹配 *.example.com
-.example.com

匹配 *example.com 和 *.*example.com
*example.com

**Reply by Mosney** [reply]
感谢解惑，另外上述回复中的三个示例后面是否应该跟着IP呢？如果不需要的话代表什么意思呢？

**Reply by PikuZheng** [reply]
> 感谢解惑，另外上述回复中的三个示例后面是否应该跟着IP呢？如果不需要的话代表什么意思呢？

就示例来说，域名前的*-和域名后的-#有不同意义。为避免混淆，域名前的*-示例没有给出域名后的内容；域名后的同理。
这里虽然有相似的符号，但是不同的意义，功能上应分开考虑，示例也应该分开。此外除了address，namesever、domain-set等也支持域名通配符。

就使用上来说，`address /xxx.com/`是无效配置，会报警告

**Reply by Mosney** [reply]
明白了，感谢。也算是个文档瑕疵
